File: Algorithm-CPP/15_소수의개수.cpp
#include <iostream>
using namespace std;

int main() {
    int n, cnt = 0;
    cin >> n;

    int arr[n + 1];
    for(int i = 0; i <= n; i++) {
        arr[i] = 0;
    }

    for(int i = 2; i <= n / 2; i++) {
        if (arr[i] == 0) {
            for(int j = i + i; j <= n; j += i) {
                arr[j] = 1;
            }
        }
    }

    for(int i = 2; i <= n; i++) {
        if (arr[i] == 0) {
            cnt++;
        }
    }

    cout << cnt << endl;

    return 0;
}

File: Algorithm-Swift/Algorithm-Swift/Inflearn/15_소수의개수.swift
//
//  15_소수의개수.swift
//  Algorithm-Swift
//
//  Created by SHIN YOON AH on 2023/07/30.
//

import Foundation

func solution15() {
    let n = Int(readLine()!)!
    var arr = Array(repeating: 1, count: n + 1)
    var cnt = 0

    for i in 2...n/2 {
        if arr[i] == 1 {
            for j in stride(from: i + i, through: n, by: i) {
                arr[j] = 0
            }
        }
    }

    for i in 2...n {
        if arr[i] == 1 {
            cnt += 1
        }
    }

    print(cnt)
}
